### PR TITLE
feat(server): query filter coresupport

### PR DESCRIPTION
**Pull request by hexaforce**
# Overview

## What I've done
In the following Project retrieval queries, only items with the coreSupport flag set to true will be returned:

* GetProjects
* GetStarredProjects
* GetDeletedProjects

Currently, projects created in the visualizer are set with coreSupport as true.

## How I tested
[e2e test](https://github.com/reearth/reearth-visualizer/pull/1231/files#diff-f7d123401a1a82a94205ef95de13237e750f27d1cce05443924ce5e3fec0e541R14)

## Memo
https://www.notion.so/eukarya/BE-Projects-query-needs-to-return-Visualizer-projects-only-12f16e0fb16580529d87c5d2a8081c5e

## Summary by CodeRabbit

- **New Features**
	- Enhanced project retrieval logic to include projects based on core support status.
	- Updated project removal process to ensure limited projects are unpublished before deletion.
	- Improved user search functionality with error handling for unmatched queries.
	- Added new test coverage for deleted projects by workspace.

- **Bug Fixes**
	- Improved filtering logic for projects to ensure accurate retrieval based on updated criteria.

- **Chores**
	- Updated dependency management for improved functionality and stability.

### CODE REVIEW

File: docker-compose.yml
Comment (pyshx) [inline docker-compose.yml:1]: Don't remove this

File: server/internal/adapter/gql/loader_user.go [hunk 1]
```diff
@@ -43,7 +44,13 @@ func (c *UserLoader) SearchUser(ctx context.Context, nameOrEmail string) (*gqlmo
...
 	}
 
-	return gqlmodel.ToUserFromSimple(res), nil
+	for _, user := range res {
+		if user.Name == nameOrEmail || user.Email == nameOrEmail {
+			return gqlmodel.ToUserFromSimple(user), nil
+		}
+	}
+
+	return nil, errors.New("user not found.")
```
Comment (pyshx) [inline server/internal/adapter/gql/loader_user.go:53]: This is not part of this PR

File: server/internal/infrastructure/memory/project.go
Comment (pyshx) [inline server/internal/infrastructure/memory/project.go:1]: Can you please write a test for this file?

**Review by pyshx — CHANGES_REQUESTED** [review]

File: docker-compose.yml
Comment (hexaforce) [reply docker-compose.yml:1]: yes

File: server/internal/infrastructure/memory/project.go
Comment (hexaforce) [reply server/internal/infrastructure/memory/project.go:1]: yes
I have added the coresupport flag to the existing test cases.

File: server/internal/adapter/gql/loader_user.go (hunk 1 above)
Comment (pyshx) [reply server/internal/adapter/gql/loader_user.go]: Ahh, Must've been from the other changes to reearthx, Please add them back in that case.

Comment (hexaforce) [reply server/internal/adapter/gql/loader_user.go:53]: yes, I reverted it!

**Review by pyshx — APPROVED** [review]

**Review by pyshx — APPROVED** [review]

**Review by pyshx — APPROVED** [review]

**Pull request merged**